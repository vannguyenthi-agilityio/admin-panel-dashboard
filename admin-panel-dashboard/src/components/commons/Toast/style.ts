import { cva } from 'class-variance-authority';

export const toastStyles = cva(
  'inline-flex items-center justify-center flex-shrink-0 rounded-lg',
  {
    variants: {
      toastType: {
        success: 'text-green-500 bg-green-500',
        warning: 'bg-yellow-500 bg-yellow-500',
        error: 'bg-red-500 border-red-500',
      },
      toastSize: {
        xs: ["h-6", "w-6", "text-xs", "px-2", "py-0"],
        sm: ["h-8", "w-8",  "text-sm", "px-2", "py-2"],
        md: ["h-12", "w-12", "text-md", "px-2", "py-2"],
        lg: ["h-14", "w-14", "text-lg", "px-2", "py-2"],
      },
    },
    defaultVariants: {
      toastType: "success",
      toastSize: "sm",
    },
  }
);
