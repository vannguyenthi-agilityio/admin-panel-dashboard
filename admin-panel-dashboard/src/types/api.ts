export type FetchFunction<T> = (onError: (error: Error) => void) => Promise<T>;
