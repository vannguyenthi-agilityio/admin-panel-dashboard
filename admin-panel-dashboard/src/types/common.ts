export interface IUser {
  name: string;
  avatar: string;
  id?: number;
}
