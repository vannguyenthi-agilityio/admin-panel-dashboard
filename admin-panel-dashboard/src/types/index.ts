export * from './icon';
export * from './button';
export * from './input';
export * from './select';
export * from './dropdown';
