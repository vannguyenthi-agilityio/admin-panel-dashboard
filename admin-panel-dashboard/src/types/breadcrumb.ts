import { type ComponentProps } from "react";
import { VariantProps } from "class-variance-authority";

import { breadcrumbStyles } from '@/components/commons/Breadcrumb/style';

export interface IItemBreadcrumb {
  href: string;
  crumbName: string;
  linkActiveCrumb: boolean;
  linkIndex: number;
  iconElement?: React.ReactNode;
}

export interface IBreadcrumb
  extends Omit<ComponentProps<"nav">, "size">,
    Omit<VariantProps<typeof breadcrumbStyles>, "iconElement">  {
  itemsBreadcrumb?: IItemBreadcrumb[];
}
