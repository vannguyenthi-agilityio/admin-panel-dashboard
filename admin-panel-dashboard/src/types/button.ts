import {  type ComponentPropsWithRef, ReactNode } from 'react';
import { type VariantProps } from 'class-variance-authority';

import { buttonStyles } from '@/components/commons/Button/style';

type ButtonElementProps = ComponentPropsWithRef<'button'>;

export interface IButton
  extends ButtonElementProps,
    VariantProps<typeof buttonStyles> {
  leftIcon?: ReactNode;
  rightIcon?: ReactNode;
  label?: string;
}
