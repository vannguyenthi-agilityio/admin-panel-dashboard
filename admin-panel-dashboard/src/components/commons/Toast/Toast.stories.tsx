import type { Meta, StoryObj } from '@storybook/react';

// Components
import Toast from './';
import {CloseIcon} from '@/components/Icons';

// Types
import { TOAST_TYPE } from '@/types';

// More on how to set up stories at: https://storybook.js.org/docs/react/writing-stories/introduction
const meta : Meta<typeof Toast> = {
  title: 'Components/Commons/Toast',
  component: Toast,
  tags: ['autodocs'],
  argTypes: {},
};

export default meta;

type Story = StoryObj<typeof meta>;

// More on writing stories with args: https://storybook.js.org/docs/react/writing-stories/args
export const Success: Story = {
  args: {
    message: "The new customer’s record is created successfully."
  },
};

export const Error: Story = {
  args: {
    toastType: TOAST_TYPE.ERROR,
    iconElement: <CloseIcon />,
    hasIconClose: true,
    message: "The new customer’s record is created error."
  },
};

export const Warning: Story = {
  args: {
    toastType: TOAST_TYPE.WARNING,
    iconElement: <CloseIcon />,
    hasIconClose: true,
    message: "The new customer’s record is created error."
  },
};
