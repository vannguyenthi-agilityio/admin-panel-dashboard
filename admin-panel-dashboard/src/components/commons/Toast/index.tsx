
import { memo, forwardRef } from "react";

// Icons
import {
  CheckIcon,
  CloseIcon
} from '@/components/Icons';

// Types
import { IToast } from "@/types";

// Helpers
import { clsxMerge } from '@/helpers';

import { toastStyles } from './style';

const Toast =  forwardRef<HTMLDivElement, IToast>(
  ({ 
    message,
    title = "Well done!",
    className,
    hasIconClose,
    toastType,
    toastSize,
    iconElement = <CheckIcon />,
    onChange,
    ...props }, ref) => (
    <div
      id={`toast-${toastType}`}
      className="flex items-start w-full max-w-xs p-4 mb-4 text-gray-500 bg-white rounded-lg shadow dark:text-gray-400 dark:bg-gray-800"
      role="alert"
      {...props}
      ref={ref}
    >
      <div
        className={clsxMerge(
        toastStyles({ toastType, toastSize }),
        className
      )}
      >
        {iconElement}
      </div>
      <div className="flex flex-col">
        <div className="ms-4 text-xl font-medium">{title}</div>
        <div className="ms-4 mt-2 text-sm font-normal">{message}</div>
      </div>
      {hasIconClose &&
        <button type="button" className="ms-auto -mx-1.5 -my-1.5 bg-white text-gray-400 hover:text-gray-900 rounded-lg focus:ring-2 focus:ring-gray-300 p-1.5 hover:bg-gray-100 inline-flex items-center justify-center h-8 w-8 dark:text-gray-500 dark:hover:text-white dark:bg-gray-800 dark:hover:bg-gray-700" data-dismiss-target="#toast-success" aria-label="Close">
          <CloseIcon stroke="#000"/>
        </button>
      }
    </div>
  )
);

Toast.displayName = "Toast";
export default memo(Toast);
