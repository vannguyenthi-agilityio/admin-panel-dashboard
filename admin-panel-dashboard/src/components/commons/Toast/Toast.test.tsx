import {
  render,
} from '@testing-library/react';
import '@testing-library/jest-dom';
import Toast from '.'

// Types
import { TOAST_TYPE } from '@/types';

describe('Toast component', () => {
  it('Renders toast success component', () => {
    const {container} = render(<Toast />)
    expect(container).toMatchSnapshot();
  });

  it('Renders toast warning component', () => {
    const {container} = render(<Toast toastType={TOAST_TYPE.WARNING} />)
    expect(container).toMatchSnapshot();
  });

  it('Renders toast error component', () => {
    const {container} = render(<Toast toastType={TOAST_TYPE.ERROR} />)
    expect(container).toMatchSnapshot();
  });
})
