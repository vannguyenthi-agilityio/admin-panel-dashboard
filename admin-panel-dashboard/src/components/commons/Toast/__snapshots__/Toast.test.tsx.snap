// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast component Renders toast error component 1`] = `
<div>
  <div
    class="flex items-start w-full max-w-xs p-4 mb-4 text-gray-500 bg-white rounded-lg shadow dark:text-gray-400 dark:bg-gray-800"
    id="toast-error"
    role="alert"
  >
    <div
      class="inline-flex items-center justify-center flex-shrink-0 rounded-lg bg-red-500 border-red-500 h-8 w-8 text-sm px-2 py-2"
    >
      <svg
        fill="none"
        height="12"
        viewBox="0 0 15 12"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1.0835 6.8665L3.67345 10.4138C4.48521 11.5256 6.1514 11.504 6.9341 10.3715L13.5835 0.75"
          stroke="#fff"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
    </div>
    <div
      class="flex flex-col"
    >
      <div
        class="ms-4 text-xl font-medium"
      >
        Well done!
      </div>
      <div
        class="ms-4 mt-2 text-sm font-normal"
      />
    </div>
  </div>
</div>
`;

exports[`Toast component Renders toast success component 1`] = `
<div>
  <div
    class="flex items-start w-full max-w-xs p-4 mb-4 text-gray-500 bg-white rounded-lg shadow dark:text-gray-400 dark:bg-gray-800"
    id="toast-undefined"
    role="alert"
  >
    <div
      class="inline-flex items-center justify-center flex-shrink-0 rounded-lg text-green-500 bg-green-500 h-8 w-8 text-sm px-2 py-2"
    >
      <svg
        fill="none"
        height="12"
        viewBox="0 0 15 12"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1.0835 6.8665L3.67345 10.4138C4.48521 11.5256 6.1514 11.504 6.9341 10.3715L13.5835 0.75"
          stroke="#fff"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
    </div>
    <div
      class="flex flex-col"
    >
      <div
        class="ms-4 text-xl font-medium"
      >
        Well done!
      </div>
      <div
        class="ms-4 mt-2 text-sm font-normal"
      />
    </div>
  </div>
</div>
`;

exports[`Toast component Renders toast warning component 1`] = `
<div>
  <div
    class="flex items-start w-full max-w-xs p-4 mb-4 text-gray-500 bg-white rounded-lg shadow dark:text-gray-400 dark:bg-gray-800"
    id="toast-warning"
    role="alert"
  >
    <div
      class="inline-flex items-center justify-center flex-shrink-0 rounded-lg bg-yellow-500 h-8 w-8 text-sm px-2 py-2"
    >
      <svg
        fill="none"
        height="12"
        viewBox="0 0 15 12"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1.0835 6.8665L3.67345 10.4138C4.48521 11.5256 6.1514 11.504 6.9341 10.3715L13.5835 0.75"
          stroke="#fff"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.5"
        />
      </svg>
    </div>
    <div
      class="flex flex-col"
    >
      <div
        class="ms-4 text-xl font-medium"
      >
        Well done!
      </div>
      <div
        class="ms-4 mt-2 text-sm font-normal"
      />
    </div>
  </div>
</div>
`;
